### DISCUSSION TITLE
Add option `--minimal-dependencies`

### DISCUSSION THREAD

**Discussion by autoantwort** [microsoft/vcpkg#25926] [Ideas]
**Is your feature request related to a problem? Please describe.**
When I install qtbase with a target triplet != host triplet, it also installs qtbase with all default features for the host system (in manifest and classic mode), but the host dependency states that only qtbase[core] is needed. So I build one hour host dependencies that I will never need.

**Proposed solution**
For host dependencies respect `"default-features": false`, for non host dependencies add an option `--minimal-dependencies` so that `"default-features": false` is even respected for the target triplet. 

**Describe alternatives you've considered**
None

**Additional context**
For cross compiling it even breaks my build because some of the dependencies does not build on the host system.  
You can workaround this issue by adding a dependency to every port that is installed for the host target and set  `"default-features": false`, but this is very tedious. 

**Previous Discussion**
https://github.com/microsoft/vcpkg/issues/11602#issuecomment-686788148

**Comment by JackBoosY** [comment]
@ras0219-msft what do you think about this?
